test/framework/gardener_utils.go

```diff
@@ -380,6 +380,15 @@ func (f *GardenerFramework) RemoveShootAnnotation(ctx context.Context, shoot *ga
 
 // MigrateShoot changes the spec.Seed.Name of a shoot and waits for it to be migrated
 func (f *GardenerFramework) MigrateShoot(ctx context.Context, shoot *gardencorev1beta1.Shoot, seed *gardencorev1beta1.Seed, prerequisites func(shoot *gardencorev1beta1.Shoot) error) error {
+	// shoot.Spec and shoot.Spec.SeedName cannot be updated at the same time, so we execute the prerequisite function here
+	if prerequisites != nil {
+		if err := f.UpdateShoot(ctx, shoot, func(shoot *gardencorev1beta1.Shoot) error {
+			return prerequisites(shoot)
```

Comment:
What about directly creating the shoot with the needed tolerations?
Updating `spec.tolerations` will trigger a reconciliation which we need to wait for, correct?

Reply:
Yeah I was wondering whether to do that. I would have to modify the shoot creation test definition, the shoot creation test and move the addition of tolerations there.
I decided to try it out like this first though and the reconciliation took less than a minute (during multiple tests) so I kept the addition here. But i'm still ok with changing it if you think it would be better in the end.